### PR TITLE
cross platform: Enable DateTime

### PR DESCRIPTION
**Pull request by obastemur**
This PR enables DateTime

### CODE REVIEW

File: lib/Runtime/Library/JavascriptNumber.h [hunk 1]
```diff
@@ -185,7 +185,7 @@ namespace Js
     // Since RecyclableObject and Int32 in their Var representation have top 14 bits 0
     // and tagged float now has top 14 XOR'd with 1s - the value with 14 bits non zero is
     // a float.
-    // A float can have all 14 bits 0 iff it was a NaN in the first place. Since
+    // A float can have all 14 bits 0 if it was a NaN in the first place. Since
```
Comment (dilijev) [inline lib/Runtime/Library/JavascriptNumber.h]: "iff" means "if and only if" -- is that not correct?

Comment (obastemur) [inline lib/Runtime/Library/JavascriptNumber.h]: ohh looked like a typo instead. wanted to fix it on the way. 

reverted this back. PR is updated.

**Comment by dilijev** [comment]
LGTM but wait for another sign off as well.

File: lib/Runtime/Base/HiResTimer.cpp [hunk 2]
```diff
@@ -17,6 +17,9 @@ namespace Js
     // bring it inline with a time server.
     double HiResTimer::GetAdjustFactor()
     {
+#ifndef _MSC_VER
```
Comment (curtisman) [inline lib/Runtime/Base/HiResTimer.cpp]: Shouldn't this be OS dependent instead of compiler dependent?

File: lib/Runtime/Base/HiResTimer.h [hunk 3]
```diff
@@ -22,7 +22,14 @@ namespace Js
 
         double GetAdjustFactor();
     public:
-        HiResTimer(): fInit(false), dBaseTime(0), baseMsCount(0),  fHiResAvailable(true), dLastTime(0), dAdjustFactor(1), fReset(true) {}
+        HiResTimer(): fInit(false), dBaseTime(0), baseMsCount(0),  
+#ifdef _MSC_VER
```
Comment (curtisman) [inline lib/Runtime/Base/HiResTimer.h]: Shouldn't this be OS dependent instead of compiler dependent?

**Comment by digitalinfinity** [comment]
LGTM as well, but echo @curtisman's comments

**Comment by obastemur** [comment]
PR is updated. @dilijev @curtisman @digitalinfinity  thanks for the review

**Comment by digitalinfinity** [comment]
@dotnet-bot test this please

**Comment by jianchun** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_debug please
@dotnet-bot test Ubuntu ubuntu_linux_release please
@dotnet-bot test Windows arm_test please

**Pull request closed without merging**